src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7420,8 +7420,11 @@ public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
             }
             else
             {
-                // Update method based on inferred receiver type: see https://github.com/dotnet/roslyn/issues/29605.
-                SetResultType(node, awaitableInfo.GetResult.ReturnTypeWithAnnotations.ToTypeWithState());
+                var reinferredGetResult = _visitResult.RValueType.Type is NamedTypeSymbol taskAwaiterType
+                    ? awaitableInfo.GetResult.OriginalDefinition.AsMember(taskAwaiterType)
+                    : awaitableInfo.GetResult;
```

Comment:
> : awaitableInfo.GetResult; [](start = 19, length = 27)

can we hit this case? Do we have a test?

Reply:
We can hit this case with some unusual scenarios such as a task-like type where the return value from GetAwaiter is a type parameter. We decided it's not worth the work to address these scenarios. I added a comment.

---
In reply to: [369191832](https://github.com/dotnet/roslyn/pull/41064#discussion_r369191832) [](ancestors = 369191832)